### ISSUE TITLE
Outset can not be built on Monterey

### ISSUE THREAD

**Issue by WardsParadox** [chilcote/outset#89]
The Requirements file for `productbuild` does not work with `productbuild` version on Monterey.

```
make pkg
rm -f ./outset*.{dmg,pkg}
pkgbuild --root pkgroot --scripts scripts --identifier com.amazon.clienteng.outset --version "3.0.3" --ownership recommended ./"outset"-"3.0.3".component.pkg
pkgbuild: Inferring bundle components from contents of pkgroot
pkgbuild: Adding top-level postinstall script
pkgbuild: Wrote package to ./outset-3.0.3.component.pkg
productbuild --identifier com.github.outset."3.0.3" --product Requirements.plist --package ./"outset"-"3.0.3".component.pkg ./"outset"-"3.0.3".pkg
2021-12-07 13:42:38.066 productbuild[51664:9316516] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[NSMutableOrderedSet unionOrderedSet:]: ordered set argument is not an NSOrderedSet'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001b3c2412c __exceptionPreprocess + 240
	1   libobjc.A.dylib                     0x00000001b3975808 objc_exception_throw + 60
	2   CoreFoundation                      0x00000001b3cf4464 -[__NSCFString characterAtIndex:].cold.1 + 0
	3   CoreFoundation                      0x00000001b3cf2eb0 -[NSMutableOrderedSet unionSet:].cold.1 + 0
	4   CoreFoundation                      0x00000001b3c055b8 -[NSMutableOrderedSet unionOrderedSet:] + 536
	5   productbuild                        0x00000001047f3acc productbuild + 31436
	6   productbuild                        0x00000001047f275c productbuild + 26460
	7   productbuild                        0x00000001047f1774 productbuild + 22388
	8   productbuild                        0x00000001047f98f4 productbuild + 55540
	9   productbuild                        0x00000001047ee8b8 productbuild + 10424
	10  dyld                                0x00000001048a90f4 start + 520
)
libc++abi: terminating with uncaught exception of type NSException
make: *** [pkg] Abort trap: 6
```

It seems like the Requirements file likely needs to be moved to a Binary plist that supports the NSOrderedSet key type.

**Comment by gregneagle** [comment]
Does `productbuild` documentation say anything about Requirements files needing to be binary plists?  I've never been aware of binary plists supporting properly list types that XML plists do not. In fact, I'm unaware of NSOrderedSet being a supported type for property lists at all. Are you aware of any documentation around this?

**Comment by WardsParadox** [comment]
NSOrderedSet is only available in Binary plists. Been seeing a large amount of projects running into this too.

Ton of details here https://github.com/xamarin/xamarin-macios/issues/13197

I'm going to be opening a feedback with apple on this as it's impacting several of our items if we use the OS restrictions

**Comment by gregneagle** [comment]
Sounds like Apple is ignoring or needs to update its own documentation.

https://developer.apple.com/documentation/foundation/nspropertylistserialization?language=objc

"A property list is itself an array or dictionary that contains only NSData, NSString, NSArray, NSDictionary, NSDate, and NSNumber objects"

**Comment by WardsParadox** [comment]
Should be resolved in 12.1
